### ISSUE TITLE
cannot download specific Qt Creator version

### ISSUE THREAD

**Issue by christian-rauch** [miurahr/aqtinstall#383]
There seems to be no option to download specific Qt Creator versions (4.15, 5.0, ...).

`aqt list-qt linux desktop` lists the Qt versions whereas `aqt list-tool linux desktop tools_qtcreator` only lists the modules (qtcreator, qtcreatordev, ...). It would be useful to also specify a specific tool version.

**Comment by ddalcino** [comment]
If you can find them on https://download.qt.io/online/qtsdkrepository/, that's what we support.

`aqt list-tool linux desktop tools_qtcreator`should list everything in the https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/tools_qtcreator/ directory; as far as I know, the Qt repo maintainers only keep one version there at a time.

**Comment by christian-rauch** [comment]
You are right, https://download.qt.io/online/qtsdkrepository/linux_x64/desktop/tools_qtcreator/qt.tools.qtcreatordev/ only lists a single (the most recent) version. The Qt Maintainance tool also provides no way to install an older version.

This might be problematic if you want to run a CI job for an older Qt Creator plugin as `aqtinstall` will then download a newer incompatible version.

I previously used https://download.qt.io/official_releases/qtcreator which provides many older Qt Creator releases. https://download.qt.io/development_releases/qtcreator/ then also provides the development releases. However, the same repo does only provide source releases of Qt and not the precompiled distribution. I guess supporting two different repo schemes is too much to ask here.

**Comment by ddalcino** [comment]
> I guess supporting two different repo schemes is too much to ask here.

A lot of our code is tightly coupled to the design of the https://download.qt.io/online/qtsdkrepository/ repo; I think it would be nice to decouple it. This repo is extremely inconsistent, and each new release of Qt adds new wrinkles and idiosyncracies. It's hard enough to support just one repo. I don't want to say we can't support another repo, but we would need a lot of help to do so.

**Comment by christian-rauch** [comment]
While I think it is very useful to only download a subset of archives of a package, I agree that the command-line interface for querying different packages, archives, etc becomes quite complicated. Right now, the CLI is already complex because `aqtinstall` allows querying different operations systems, architectures and versions. Overall, I see why you are reluctant to add this functionality for users.

In the end, I did not use `aqtinstall` because I usually only want a givens version of Qt and Qt Creator for the operating system and architecture the script is running on. The implementation of this custom tool is quite small and it achieves what I need.

@ddalcino The short answer to your question
> Would this proposed tool be useful to anyone?

is: Yes, I believe such a tool would be useful because I would have used it if the functionality would have been available. But I am not going to use it anymore, now that I implemented a custom script that does exactly what I need with less amount of code :-)
